File: duplicate-assert/java-junit/TestURIHandlerService_testGetAuthorityWithScheme.java
public class TestURIHandlerService extends XTestCase {
    @Test
    public void testGetAuthorityWithScheme() throws Exception {
        URIHandlerService uriService = new URIHandlerService();
        URI uri = uriService.getAuthorityWithScheme("hdfs://nn1:8020/dataset/${YEAR}/${MONTH}");
        assertEquals("hdfs://nn1:8020", uri.toString());
        uri = uriService.getAuthorityWithScheme("hdfs://nn1:8020");
        assertEquals("hdfs://nn1:8020", uri.toString());
        uri = uriService.getAuthorityWithScheme("hdfs://nn1:8020/");
        assertEquals("hdfs://nn1:8020", uri.toString());
        uri = uriService.getAuthorityWithScheme("hdfs://///tmp/file");
        assertEquals("hdfs:///", uri.toString());
        uri = uriService.getAuthorityWithScheme("hdfs:///tmp/file");
        assertEquals("hdfs:///", uri.toString());
        uri = uriService.getAuthorityWithScheme("/tmp/file");
        assertEquals("/", uri.toString());
    }
}

File: duplicate-assert/CSharpXUnit/CSharpXUnit.Tests/Tests/TestURIHandlerService_testGetAuthorityWithScheme.cs
public class TestURIHandlerService {
    [Fact]
    public void TestGetAuthorityWithScheme() {
        var uriService = new URIHandlerService();
        var uri = uriService.GetAuthorityWithScheme("hdfs://nn1:8020/dataset/${YEAR}/${MONTH}");
        Assert.Equal("hdfs://nn1:8020", uri.ToString());
        uri = uriService.GetAuthorityWithScheme("hdfs://nn1:8020");
        Assert.Equal("hdfs://nn1:8020", uri.ToString());
        uri = uriService.GetAuthorityWithScheme("hdfs://nn1:8020/");
        Assert.Equal("hdfs://nn1:8020", uri.ToString());
        uri = uriService.GetAuthorityWithScheme("hdfs://///tmp/file");
        Assert.Equal("hdfs:///", uri.ToString());
        uri = uriService.GetAuthorityWithScheme("hdfs:///tmp/file");
        Assert.Equal("hdfs:///", uri.ToString());
        uri = uriService.GetAuthorityWithScheme("/tmp/file");
        Assert.Equal("/", uri.ToString());
    }
}
